### PR TITLE
Use Host in Query Parameter if allowed

### PR DESCRIPTION
**Pull request by pfz4**
This Commit allows Admins to add a Host to the connect URL (https://example.com?host=127.0.0.1:3389).
This way Admins can link to Remote Desktop Sessions on a specific Host in Links.

Only works, if the host query Parameter is in the Host List in the Config. Otherwise, the Server picks a random Host.

### CODE REVIEW

File: api/web.go [hunk 1]
```diff
@@ -157,6 +157,17 @@ func (c *Config) HandleDownload(w http.ResponseWriter, r *http.Request) {
 	rand.Seed(time.Now().Unix())
 	host := c.Hosts[rand.Intn(len(c.Hosts))]
 	host = strings.Replace(host, "{{ preferred_username }}", userName, 1)
+	
+	
+	//Change Host to value of host query Parameter if host query string is in Config Hosts
+	paraHost, hasParaHost := r.URL.Query()["host"]
```
Comment (bolkedebruin) [inline api/web.go:163]: Can you write out the whole name of the variable. “paramHost” and “parameterHost” are acceptable.

File: api/web.go [hunk 2]
```diff
@@ -157,6 +157,17 @@ func (c *Config) HandleDownload(w http.ResponseWriter, r *http.Request) {
...
+	if hasParaHost {
+        for _,a := range c.Hosts {
+            if a==paraHost[0] {
```
Comment (bolkedebruin) [inline api/web.go:166]: Please lint your code

File: api/web.go [hunk 3]
```diff
@@ -157,6 +157,17 @@ func (c *Config) HandleDownload(w http.ResponseWriter, r *http.Request) {
 	rand.Seed(time.Now().Unix())
 	host := c.Hosts[rand.Intn(len(c.Hosts))]
 	host = strings.Replace(host, "{{ preferred_username }}", userName, 1)
+	
+	
+	//Change Host to value of host query Parameter if host query string is in Config Hosts
+	paraHost, hasParaHost := r.URL.Query()["host"]
+	if hasParaHost {
+        for _,a := range c.Hosts {
```
Comment (bolkedebruin) [inline api/web.go:165]: Please lint your code.

Comment (bolkedebruin) [inline api/web.go:165]: Please use a more descriptive variable name.

**Review by bolkedebruin — CHANGES_REQUESTED** [review]
Can you add a test to your code please?

File: api/web.go (hunk 3 above)
Comment (pfz4) [reply api/web.go]: Lint hasn't suggested changes in new version

File: api/web.go (hunk 2 above)
Comment (pfz4) [reply api/web.go]: Lint hasn't suggested changes in new version

**Comment by pfz4** [comment]
Build check failed because of go.sum at that time. Works on my PC and my Gitlab Pipeline.

**Comment by pfz4** [comment]
@bolkedebruin is it possible to restart the Pipeline? I'm pretty sure it's a missing dependency (that I've not introduced) and not my code.

**Comment by bolkedebruin** [comment]
If you rebase and push again the pipeline will start itself. Please address the comments above

**Comment by pfz4** [comment]
@bolkedebruin The changes you requested are implemented. Go Lint shows no suggestion in the area I edited.

**Comment by bolkedebruin** [comment]
@pfz4 I am not seeing any test so that this works as intended?

**Comment by pfz4** [comment]
@bolkedebruin Sorry for taking so long. Is this procedure correct?
1. Move my Code to its own function
2. Create a `cmd/rdpgw/api/web_test.go`
3. Create a TestCase to test the function if the correct Host is returned, a random when a wrong host is selected or no host is specified

**Comment by bolkedebruin** [comment]
This has been solved in master now, albeit a bit different. See code/docs.

**Pull request closed without merging**